k2/python/csrc/torch/index_add.cu

```diff
@@ -22,12 +24,15 @@ namespace k2 {
 
 // It implements  (*in_out)[index[i]] += value[i]
 // for i in range(index.numel())
-void IndexAdd(torch::Tensor index, torch::Tensor value, torch::Tensor *in_out) {
-  // We support only 1-D tensors at present
+static void IndexAdd1D(torch::Tensor index, torch::Tensor value,
+                       torch::Tensor *in_out) {
+  NVTX_RANGE(K2_FUNC);
   K2_CHECK_EQ(index.scalar_type(), ToScalarType<int32_t>::value);
   K2_CHECK_EQ(index.dim(), 1);
 
-  K2_CHECK_EQ(value.scalar_type(), ToScalarType<float>::value);
+  K2_CHECK_EQ(value.scalar_type(), ToScalarType<float>::value)
+      << "index_add is supposed to be used in back propagation, "
```

Comment:
I'd rather just say something to the effect that only float is supported right now.  It's a general purpose function.

Reply:
Fixed by removal.